processor/src/main/kotlin/com/tobrun/datacompat/DataCompatProcessor.kt

```diff
@@ -233,7 +230,24 @@ class DataCompatProcessor(
                         propertyMap.keys.joinToString(
                             prefix = "return ",
                             separator = "·&& ",
-                            transform = { "$it·==·other.$it" },
+                            transform = {
+                                val resolvedType = it.type.resolve()
+                                val isFloat =
+                                    resolvedType.toClassName() == Float::class.asTypeName()
+                                val isDouble =
+                                    resolvedType.toClassName() == Double::class.asTypeName()
```

Comment:
```suggestion
                                val isFloatOrDouble =
                                    resolvedType.toClassName() in setOf(Float::class.asTypeName(),
Double::class.asTypeName())
```

Reply:
I have separate checks for `isMarkedNullable && isDouble` and `isMarkedNullable && isFloat` so I need separate variables.